packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js

```diff
@@ -13,29 +13,47 @@ import Carousel from '../ProductImageCarousel';
 import { fullPageLoadingIndicator } from '../LoadingIndicator';
 import Quantity from '../ProductQuantity';
 import RichText from '../RichText';
-import CREATE_CART_MUTATION from '../../queries/createCart.graphql';
-import GET_CART_DETAILS_QUERY from '../../queries/getCartDetails.graphql';
+
 import defaultClasses from './productFullDetail.css';
 import {
     ADD_CONFIGURABLE_MUTATION,
     ADD_SIMPLE_MUTATION
 } from './productFullDetail.gql';
-
+import CREATE_CART_MUTATION from '../../queries/createCart.graphql';
 const Options = React.lazy(() => import('../ProductOptions'));
 
+// Correlate a GQL error message to a field. GQL could return a longer error
+// string but it may contain contextual info such as product id. We can use
+// parts of the string to check for which field to apply the error.
+const ERROR_MESSAGE_TO_FIELD_MAP = {
+    'The requested qty is not available': 'quantity',
+    'Product that you are trying to add is not available.': 'quantity',
+    "The product that was requested doesn't exist.": 'quantity',
+
+    // These errors cause a retry within the action so we don't want to render
+    // an "unknown" form error as it will appear and the disappear.
+    'Could not find a cart': 'user',
```

Comment:
I see why you did this; so the form error isn't rendered on these, but this indicates to me there's a user field. This should solve itself when we pull retry logic, just making a note for myself for the re-review.